src/Component/RequestBuilder/QueryElevation.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Solarium\Component\RequestBuilder;
+
+use Solarium\Component\QueryElevation as QueryelevationComponent;
+use Solarium\Core\Client\Request;
+
+/**
+ * Add select component queryelevation to the request.
+ */
+class QueryElevation implements ComponentRequestBuilderInterface
+{
+    /**
+     * Add request settings for QueryelevationComponent.
+     *
+     * @param QueryelevationComponent $component
+     * @param Request                 $request
+     *
+     * @return Request
+     */
+    public function buildComponent($component, $request)
+    {
+        // switch request handler
+        $request->setHandler($component->getHandler());
```

Comment:
A component should never secretly switch the request handler. This might break the main query.